core/src/config/transaction_extensions.rs

```diff
@@ -582,12 +582,13 @@ macro_rules! impl_tuples {
             ) -> Result<Self, ExtrinsicParamsError> {
                 let metadata = &client.metadata;
                 let types = metadata.types();
+                let tx_extension_version = metadata.extrinsic().transaction_extension_version_to_use_for_encoding();
 
                 // For each transaction extension in the tuple, find the matching index in the metadata, if
                 // there is one, and add it to a map with that index as the key.
                 let mut exts_by_index = HashMap::new();
                 $({
-                    for (idx, e) in metadata.extrinsic().transaction_extensions().iter().enumerate() {
+                    for (idx, e) in metadata.extrinsic().transaction_extensions_by_version(tx_extension_version).unwrap().enumerate() {
```

Comment:
Maybe add an expect here or explain that this may fail if an `invalid transaction version`?

Or add some docs on the macro itself...